its/ruling/src/test/resources/expected/python-S5806.json

```diff
@@ -0,0 +1,2049 @@
+{
```

Comment:
isn't it a bit too noisy? I think it may be worth checking with @nicolas-harraudeau-sonarsource if we can avoid some of those issues

Reply:
I share the feeling. It appears that some builtin names are quite popular names: `max, id, dict, type`, etc...Not sure what should be done: those are indeed TP, developers seem to be comfortable with shadowing builtins.

Reply:
I would like to see on peach if all projects are like this or if only a few of them do this. Personally I wouldn't remove any builtin from the list. Shadowing builtins can be a source of bugs in case of refactoring. Ex: you rename the variable but forget a few places and they use the builtin instead. There is also a simple solution: add an underscore at the end of the name or find another name.

There is however one thing which came to my mind this morning. We should not raise issues on parameters of methods overriding a method from a parent class. Developers can't change these parameters or they would break the Liskov substitution principle. These issues should only appear on the top overriden method. If we don't have the full class hierarchy we would not raise at all just to avoid FPs. Does it make sense @dmitry-golovinov-sonarsource ?

Reply:
Another solution would be to have a parameter enabling the rule on method and function parameters, and it would be disabled by default. This would avoid raising on parameter names when developers have to follow an API (ex: for duck typing).

Reply:
As discusser IRL I'll modify the RSPEC as follows:
* I will add a parameter enabling the validation of function/method/lambda parameters. It will be disabled, i.e. set to False, by default.
* I will document in the RSPEC description the exception about function/method/lambda parameters.

We also agreed that when the parameter is enabled we won't raise on overriding methods.